src/log.rs

```diff
@@ -58,3 +60,39 @@ macro_rules! emit_event {
         $ctx.emit_event($event);
     };
 }
+
+pub trait LogExt<T> {
+    /// Emits a warning if the receiver contains an Err value.
+    ///
+    /// Returns an [`Option<T>`] with the `Ok(_)` value, if any:
+    /// - You won't get any warnings about unused results but can still use the value if you need it
+    /// - This prevents the same warning from being printed to the log multiple times
+    ///
+    /// Thanks to the [track_caller](https://blog.rust-lang.org/2020/08/27/Rust-1.46.0.html#track_caller)
+    /// feature, the location of the caller is printed to the log, just like with the warn!() macro.
+    #[track_caller]
+    fn log(self, context: &Context) -> Option<T>;
```

Comment:
this seems to be mostly the same than `log_err()` in the ffi part. maybe better use this name then - it is already known and describes better what happens - log something in case of errors.

Reply:
I also thought about this; the reasons why I decided against it:
- `log_err()` might make people think that an error is logged, while actually a warning is logged
- a possible name collision: If we call this function here `log_err()`, we can only use this function or the existing function in the ffi. And they don't behave exactly the same, `log()` returns an Option, `log_err()` a Result. Not sure whether this might be a problem.
  BTW I also thought about calling it `log_ok()` as it does the same as `log_err().ok()` but then assumed that nobody would understand what the `ok` means.
- well, it's shorter :) not a big thing now, but we may want to create another function `log_msg(self, context, string)` or `log_s(self, context, string)` or similar that can be passed an extra string message. `log_err_s()` and `log_err_msg()` seem to be confusingly long then.

Reply:
wrt warning/error - i think this that part is fine, it logs returned Errors - as warnings, as we do for all non-user-visible errors.

however, regarding all the potential troubles, i would also be fine with just `log()` :) this is no blocker or so :)